src/bin/test_serialize.rs

```diff
@@ -0,0 +1,73 @@
+use cattus::chess::chess_game::ChessPosition;
+use cattus::chess::net::serializer::ChessSerializer;
+use cattus::game::common::{GameColor, GamePosition, IGame};
+use cattus::game::self_play::DataSerializer;
+use cattus::hex::hex_game::HexPosition;
+use cattus::hex::net::serializer::HexSerializer;
+use cattus::ttt::net::serializer::TttSerializer;
+use cattus::ttt::ttt_game::TttPosition;
+use clap::Parser;
+use itertools::Itertools;
+
+#[derive(Parser, Debug)]
+#[clap(about, long_about = None)]
+struct Args {
+    #[clap(long)]
+    game: String,
+    #[clap(long)]
+    position: String,
+    #[clap(long)]
+    outfile: String,
+}
+
+fn main() -> std::io::Result<()> {
+    let args = Args::parse();
+    match args.game.as_str() {
+        "tictactoe" => test_tictactoe(args),
+        "hex" => test_hex(args),
+        "chess" => test_chess(args),
+        unknown_game => panic!("unknown game: {:?}", unknown_game),
+    }
+}
+
+fn test_tictactoe(args: Args) -> std::io::Result<()> {

```

Comment:
why did you prefer to duplicate these instead of putting them inside `main`?

Reply:
duplication is necessary because from_str is currently not in the trait API. in the main func itself or outside it doesn't matter